### PR TITLE
Pull Request for Issue2078: Creating beamline support class with convenience methods.

### PR DESCRIPTION
**Pull request by helfrij**
Created new namespace BioXASBeamlineSupport that contains misc helper methods related to matching beamline components to detectors, data sources, scans, and configurations. These methods are used in BioXASBeamline and in BioXASXASScanActionController, simplifying both. Also, noticed that the dark current corrected sources for the diode, PIPS, and Lytle detectors were missing, should have been included in #2044. Added them here.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you take a look at this request?

**Comment by XuetaoLiu-CLS** [comment]
Looks good.

Time Estimates: 10 minutes

File: source/acquaman/BioXAS/BioXASXASScanActionController.cpp [hunk 1]
```diff
@@ -50,82 +50,55 @@ void BioXASXASScanActionController::createScanAssembler()
...
-			zebraTriggerSource->addDetectorManager(scaler);
-	}
+	AMDataSource *dwellTimeSource = BioXASBeamlineSupport::scalerDwellTimeDetectorSource(scan_);
 
 	// Create analyzed data source for the absorbance.
 
 	AM1DExpressionAB *absorbanceSource = 0;
 
 	if (i0DetectorSource && i1DetectorSource && i2DetectorSource) {
+		qDebug() << "BioXASXASScanActionController: the scan is using I0, I1, and I2. Creating absorbance source.";
```
Comment (dretrex) [inline source/acquaman/BioXAS/BioXASXASScanActionController.cpp]: Do you still need this debug?

Comment (helfrij) [inline source/acquaman/BioXAS/BioXASXASScanActionController.cpp]: Nope. I'll remove.

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 10 minutes

**Comment by helfrij** [comment]
@dretrex merge conflicts fixed here.

**Pull request merged**